ch5/ex-5-13-tail.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Exercise 5-13
+ *
+ * Write the program tail, which prints the last n lines of its input.  By
+ * default, n is 10, let us say, but it can be changed by an optional
+ * argument, so that
+ *
+ * 	tail -n
+ *
+ * prints the last n lines.  The program should behave rationally no matter
+ * how unreasonable the input or the value of n.  Write the program so it
+ * makes the best use of available storage; lines should be stored as in the
+ * sorting program of Section 5.6, not in a two-dimensional array of fixed
+ * size.
+ */
+
+#include <ctype.h>
+#include <limits.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define INITIAL_BUFFSIZE (1 << 10)
+#define DEFAULT_TAILSIZE 10
+
+enum { OK, ERROR };
```

Comment:
Since these values are being compared directly to `0` it might be a good idea to make it explicit like this ` enum { OK=0, ERROR };`.
There is also `stdbool.h` which defines `true` and `false` if that appeals.

Reply:
I don't think they are compared to `0` in this case, but some _random idiot™_ might come along and assume they can be - will change.  

stdbool's not in the toolbox yet.